File: app/src/main/java/dev/olog/msc/presentation/widget/DottedSeparator.java
package dev.olog.msc.presentation.widget;

import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;

import dev.olog.msc.R;

public class DottedSeparator extends View {

    public DottedSeparator(Context context) {
        this(context, null);
    }

    public DottedSeparator(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public DottedSeparator(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        this(context, attrs, defStyleAttr, 0);
    }

    public DottedSeparator(Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        setLayerType(View.LAYER_TYPE_SOFTWARE, null);
        setAlpha(.1f);
        setBackgroundResource(R.drawable.dotted_line);
    }
}

File: features/feature-presentation-base/src/main/java/dev/olog/feature/presentation/base/widget/DottedSeparator.kt
package dev.olog.feature.presentation.base.widget

import android.content.Context
import android.content.res.ColorStateList
import android.graphics.Color
import android.util.AttributeSet
import android.view.View
import dev.olog.feature.presentation.base.R

class DottedSeparator(
    context: Context,
    attrs: AttributeSet
) : View(context, attrs) {

    init {
        val isDarkMode = context.resources.getBoolean(R.bool.is_dark_mode)
        if (!isDarkMode) {
            setLayerType(LAYER_TYPE_SOFTWARE, null)
            setBackgroundResource(R.drawable.dotted_line)
            backgroundTintList = ColorStateList.valueOf(Color.BLACK);
            alpha = .1f;
        }
    }
}